capsules/extra/src/test/siphash24.rs

```diff
@@ -56,10 +59,33 @@ impl Client<8> for TestSipHash24 {
     fn add_data_done(&self, _result: Result<(), ErrorCode>, _data: SubSlice<'static, u8>) {}
 
     fn hash_done(&self, _result: Result<(), ErrorCode>, digest: &'static mut [u8; 8]) {
-        debug!("hashed result:   {:?}", digest);
-        debug!("expected result: {:?}", self.correct_hash.take().unwrap());
+        let correct = self.correct_hash.take().unwrap();
+
+        let mut matches = true;
+        for i in 0..8 {
+            if correct[i] != digest[i] {
+                matches = false;
+                kernel::debug!("TestSipHash24: incorrect hash output!");
+            }
+        }
```

Comment:
```suggestion
        let matches = correct != digest;
        if !matches {
            kernel::debug!("TestSipHash24: incorrect hash output!");
        }
```
You can just compare equal length arrays directly!